crates/alloc/src/stats.rs

```diff
@@ -87,58 +87,56 @@ pub fn collect_jemalloc_stats() -> Result<JemallocStats, Error> {
 
 #[cfg(feature = "metrics")]
 pub fn update_jemalloc_metrics() -> Result<(), Error> {
-    use metrics::{gauge, otel};
+    use metrics::gauge;
 
     let stats = collect_jemalloc_stats()?;
     let total = &stats.total;
 
     // Total number of bytes allocated by the application. This corresponds to
     // `stats.allocated` in jemalloc's API.
-    gauge!("jemalloc_memory_allocated", total.allocated);
+    gauge!("jemalloc_memory_allocated").set(total.allocated as f64);
 
     // Total number of bytes in active pages allocated by the application. This
     // corresponds to `stats.active` in jemalloc's API.
-    gauge!("jemalloc_memory_active", total.active);
+    gauge!("jemalloc_memory_active").set(total.active as f64);
 
     // Total number of bytes dedicated to `jemalloc` metadata. This corresponds to
     // `stats.metadata` in jemalloc's API.
-    gauge!("jemalloc_memory_metadata", total.metadata);
+    gauge!("jemalloc_memory_metadata").set(total.metadata as f64);
 
     // Total number of bytes in physically resident data pages mapped by the
     // allocator. This corresponds to `stats.resident` in jemalloc's API.
-    gauge!("jemalloc_memory_resident", total.resident);
+    gauge!("jemalloc_memory_resident").set(total.resident as f64);
 
     // Total number of bytes in active extents mapped by the allocator. This
     // corresponds to `stats.mapped` in jemalloc's API.
-    gauge!("jemalloc_memory_mapped", total.mapped);
+    gauge!("jemalloc_memory_mapped").set(total.mapped as f64);
 
     // Total number of bytes in virtual memory mappings that were retained rather
     // than being returned to the operating system via e.g. `munmap(2)`. This
     // corresponds to `stats.retained` in jemalloc's API.
-    gauge!("jemalloc_memory_retained", total.retained);
+    gauge!("jemalloc_memory_retained").set(total.retained as f64);
 
     let bin_const = stats.arena_constants.bin.iter();
     let bin_stats = stats.arena_stats.merged.bins.iter();
 
     for (bin_const, bin_stats) in bin_const.zip(bin_stats) {
-        let tags = [otel::KeyValue::new(
-            "bin_size",
-            bin_const.size.try_into().unwrap_or(0i64),
-        )];
+        let gauge =
```

Comment:
Why is this now a function instead of using macro like it was before?

Reply:
Just DRY, we could use macro is well